pkg/actuators/machine/instances_test.go

```diff
@@ -55,6 +55,37 @@ func TestRemoveDuplicatedTags(t *testing.T) {
 	}
 }
 
+func TestBuildTagList(t *testing.T) {
+	cases := []struct {
+		tagList  []awsproviderv1.TagSpecification
+		expected []*ec2.Tag
+	}{
+		{
+			tagList: []awsproviderv1.TagSpecification{},
+			expected: []*ec2.Tag{
+				{Key: aws.String("kubernetes.io/cluster/clusterID"), Value: aws.String("owned")},
+				{Key: aws.String("Name"), Value: aws.String("machineName")},
+			},
+		},
+		{
+			tagList: []awsproviderv1.TagSpecification{
+				{Name: "Name", Value: "badname"},
+				{Name: "kubernetes.io/cluster/badid", Value: "badvalue"},
+			},
+			expected: []*ec2.Tag{
+				{Key: aws.String("kubernetes.io/cluster/clusterID"), Value: aws.String("owned")},
+				{Key: aws.String("Name"), Value: aws.String("machineName")},
+			},
+		},
```

Comment:
Let's add an extra value to this so that we can see good values are preserved
```suggestion
		{
			tagList: []awsproviderv1.TagSpecification{
				{Name: "Name", Value: "badname"},
				{Name: "kubernetes.io/cluster/badid", Value: "badvalue"},
				{Name: "good", Value: "goodvalue"},
			},
			expected: []*ec2.Tag{
				{Key: aws.String("kubernetes.io/cluster/clusterID"), Value: aws.String("owned")},
				{Key: aws.String("Name"), Value: aws.String("machineName")},
				{Key: aws.String("good"), Value: aws.String("goodvalue")},
			},
		},
```